S/SHTOOLS/build_tarballs.jl

```diff
@@ -14,19 +14,28 @@ script = raw"""
 cd $WORKSPACE/srcdir/SHTOOLS-*
 
 # Patch source code
-# Add missing C interface for MakeGradientDH
-atomic_patch -p1 $WORKSPACE/srcdir/patches/add-cMakeGradientDH.patch
 # Don't use libtool
 atomic_patch -p0 $WORKSPACE/srcdir/patches/no-libtool.patch
+# Correct C interface for MakeGradientDH (see <https://github.com/SHTOOLS/SHTOOLS/issues/328>)
+atomic_patch -p1 $WORKSPACE/srcdir/patches/correct-cMakeGradientDH.patch
 
 # Build and install static libraries
 make fortran -j${nproc} F95FLAGS="-fPIC -O3 -std=gnu"
 make fortran-mp -j${nproc} F95FLAGS="-fPIC -O3 -std=gnu"
 make install PREFIX=${prefix}
 
 # Create shared libraries
-gfortran -shared -o ${libdir}/libSHTOOLS.${dlext} -Wl,$(flagon --whole-archive) ${prefix}/lib/libSHTOOLS.a -Wl,$(flagon --no-whole-archive | cut -d' ' -f1) -lfftw3 -lopenblas -lm
-gfortran -fopenmp -shared -o ${libdir}/libSHTOOLS-mp.${dlext} -Wl,$(flagon --whole-archive) ${prefix}/lib/libSHTOOLS-mp.a -Wl,$(flagon --no-whole-archive | cut -d' ' -f1) -lfftw3 -lopenblas -lm
+# TODO: Do we need to add " | cut -d' ' -f1"?
```

Comment:
I thought I fixed that in #2988

Reply:
My assumption was that, if the package builds for all architectures, then this is fixed.

Reply:
It worked.